src/rt/cast_.d

```diff
@@ -13,6 +13,15 @@
  */
 module rt.cast_;
 
+// because using == does a dynamic cast, but we
+// are trying to implement dynamic cast.
```

Comment:
Since this is comparing for equality, the name for it should be something like "equals".